日本語ロケールを対応

st1t opened the pull request:
# 修正内容

- 日本語ロケールを対応
- github actionsにrubocopを追加

# スクリーンショット

## 日本語画面

![image](https://user-images.githubusercontent.com/16526902/107616331-2f075700-6c91-11eb-95f5-c6079bd6f3c2.png)
---
![image](https://user-images.githubusercontent.com/16526902/107616513-932a1b00-6c91-11eb-818a-7ed2c479647b.png)
---
![image](https://user-images.githubusercontent.com/16526902/107616535-a0470a00-6c91-11eb-96c9-d5c0a765ce95.png)

## 英語画面

![image](https://user-images.githubusercontent.com/16526902/107616445-72fa5c00-6c91-11eb-9ab1-2c0bb665a659.png)

## rubocop

![image](https://user-images.githubusercontent.com/16526902/107616370-43e3ea80-6c91-11eb-81a4-3ecfc8380df0.png)

st1t committed:
Add Japanese locale

st1t committed:
Add github actions

st1t committed:
Modify github actions

torinoko commented on app/views/books/_form.html.erb:
```diff
@@ -12,26 +12,26 @@
 
   <div class="field">
-    <%= form.label :title %>
+    <%= form.label t(:'book.title') %>
```
`t` の引数はシンボルにせず、 `t('book.title')` で大丈夫ですよ〜！

st1t replied:
ありがとうございます！
別のコミットで結果的には上書きしてしまっていますが、 https://github.com/st1t/fjord-books_app/pull/1/commits/654e8d8306a493157918c9a95e8f9daf775308ad で`t`の引数の書き方は統一しました🙏

torinoko commented on app/views/books/edit.html.erb:
```diff
@@ -1,6 +1,6 @@
-<h1>Editing Book</h1>
+<h1><%= t(:'sentence.editing_book') %></h1>
```
[Active Recordモデルで翻訳を行なう](https://railsguides.jp/i18n.html#active-record%E3%83%A2%E3%83%87%E3%83%AB%E3%81%A7%E7%BF%BB%E8%A8%B3%E3%82%92%E8%A1%8C%E3%81%AA%E3%81%86)
このあたりのメソッドを活用してもいいかもしれませんね。

あと "Editing" と "Book（モデル名）" は別々に翻訳するという手もありますね。
今回はモデルが Book のみですが、モデルが増えて Editing User、Editing Order、Editing xxx... とページも増えてきたとき、すべてに対して `sentence.editing_model` の翻訳を用意するのは大変そうな気がしませんか？

st1t replied:
> このあたりのメソッドを活用してもいいかもしれませんね。

ありがとうございます、確かにActive Recordモデルで翻訳を行ったほうが良さそうですね・・・！

> あと "Editing" と "Book（モデル名）" は別々に翻訳するという手もありますね。
> 今回はモデルが Book のみですが、モデルが増えて Editing User、Editing Order、Editing xxx... とページも増えてきたとき、すべてに対して sentence.editing_model の翻訳を用意するのは大変そうな気がしませんか？

言われてみれば先行きが暗すぎるコードでした💦
この辺りは https://github.com/st1t/fjord-books_app/pull/1/commits/283dd21ec3e4e1a74009ec33881be073aef7b02b でActive Recordモデルで翻訳するものと一緒に対応してみました

そして追加したあとに文法による順序の問題に気がついたので、 https://github.com/st1t/fjord-books_app/pull/1/commits/bcb810c8ed239e3d32ef9bb8e97d33d3426eb5b1 を追加してます🙏

torinoko reviewed (commented):
@st1t 
PR の書き方やコミットの粒度、メッセージが素晴らしいですね！✨

気になった点をコメントしました。
あと必須要件にある英語の YAML ファイルがないようなのでご確認いただけますか🙏
> 英語と日本語の翻訳をconfig/locales配下のYAMLファイルに定義する

st1t committed:
Modify the format of the translate method arguments to make them consistent in views/books

st1t committed:
Modify to translate in Active Record model

- Modify ja.yml to support Active Record structure
- Modify books view to use Model.model_name.human and Model.human_attribute_name method

st1t committed:
Add english translation information to en.yml

st1t committed:
Modify locale to match the grammar

st1t commented on app/views/books/index.html.erb:
```diff
@@ -1,14 +1,14 @@
 <p id="notice"><%= notice %></p>
 
-<h1>Books</h1>
+<h1><%= Book.model_name.human(count:@books.count) %></h1>
```
ここはcountで表示を出し分けなくても`書籍一覧`で一貫して表示してもいいよなぁと思いつつ、この書き方を知ったので仕事でもないですしせっかくなので試してみた、くらいの書き方になります。

torinoko replied:
お！挑戦いいですね〜！ 👍

st1t commented on config/locales/en.yml:
```diff
@@ -30,4 +30,22 @@
 
 en:
-  hello: "Hello world"
+  activerecord:
```
すいません、`en.yml`に情報を追加しました💦

st1t reviewed (commented):
@torinoko
ご確認ありがとうございます！
英語のYAML、すいません、完全に読み違えてました💦
改めてご確認お願いいたします🙏

torinoko commented on app/views/books/_form.html.erb:
```diff
@@ -12,26 +12,26 @@
 
   <div class="actions">
-    <%= form.submit %>
+    <%= form.submit t('action.add_object', obj: Book.model_name.human) %>
```
引数を渡して表示する書き方もマスターですね！✨

torinoko reviewed (approved):
@st1t 
とてもよいと思います〜！✨
修正内容もしっかりしていますが、コミットメッセージや粒度が本当に綺麗で惚れ惚れしますね。すごいです〜！
